ui/src/pages/insightDetail/components/eventAggregator/index.tsx

```diff
@@ -401,42 +405,46 @@
             />
           </Space>
         </div>
-        <div className={styles.events_content_diagnosisContent}>
-          {diagnosisStatus === 'loading' ||
-          (diagnosisStatus === 'streaming' && !diagnosis) ? (
-            <div className={styles.events_content_diagnosisLoading}>
-              <Spin />
-              <p>{t('EventAggregator.DiagnosisInProgress')}</p>
-            </div>
-          ) : diagnosisStatus === 'error' ? (
-            <Alert
-              type="error"
-              message={t('EventAggregator.DiagnosisFailed')}
-              description={t('EventAggregator.TryAgainLater')}
-            />
-          ) : (
-            <>
-              <Markdown>{diagnosis}</Markdown>
-              {diagnosisStatus === 'streaming' && (
-                <div className={styles.events_content_streamingIndicator}>
-                  <span className={styles.dot}></span>
-                  <span className={styles.dot}></span>
-                  <span className={styles.dot}></span>
-                </div>
-              )}
-              <div ref={diagnosisEndRef} />
-            </>
-          )}
+        <div className={styles.events_content_diagnosisBody}>
+          <div
+            className={styles.events_content_diagnosisContent}
+            ref={contentRef}
+          >
+            {diagnosisStatus === 'loading' ||
+            (diagnosisStatus === 'streaming' && !diagnosis) ? (
+              <div className={styles.events_content_diagnosisLoading}>
+                <Spin />
+                <p>{t('EventAggregator.DiagnosisInProgress')}</p>
+              </div>
+            ) : diagnosisStatus === 'error' ? (
+              <Alert
+                type="error"
+                message={t('EventAggregator.DiagnosisFailed')}
+                description={t('EventAggregator.TryAgainLater')}
+              />
+            ) : (
+              <>
+                <Markdown>{diagnosis}</Markdown>
+                {diagnosisStatus === 'streaming' && (
+                  <div
+                    className={`${styles.events_content_streamingIndicator} ${dotToTopHeight - contentToTopHeight + 53 - (isVertical ? 300 : 600) >= 0 ? styles.events_content_streamingIndicatorFixed : ''}`}
```

Comment:
The condition for adding the `fixed` class is quite complex. Consider breaking it into smaller variables for better readability and maintainability.